Dockerfile

```diff
@@ -0,0 +1,35 @@
+# FROM node:8-alpine also works here for a smaller image
+FROM node:8-alpine
+
+# set our node environment, either development or production
+# defaults to production, compose overrides this to development on build and run
+ENV NODE_ENV production
+
+EXPOSE 9292 9293
+
+# you'll likely want the latest npm, regardless of node version, for speed and fixes
+# but pin this version for the best stability
+RUN npm i npm@latest -g
+
+# install dependencies first, in a different location for easier app bind mounting for local development
+WORKDIR /opt
+COPY package.json package-lock.json* ./
+RUN npm install --no-optional && npm cache clean --force
```

Comment:
DockerFiles are used during building the docker image, which is a design time CI/CD activity. So, I am not sure if there would be any issue. Let's keep the private cloud part aside as we do not know how it would shape up.

Regarding whether to keep the DockerFile here or not, because during building of the image, you need it on the root of the src directory, hence it is better to have it here IMO. Also, the interoperator docker file is autogenerated.

Reply:
I would rather do it the following way:
- Have the CI/CD tool (e.g. Nexus) run the npm install for pulling in all dependencies and creating a tarball
- The Dockerfile can unzip the tarball and run the required "copy" steps as required

Reply:
As discussed, I don't have a problem with keeping the Dockerfile in the repository. The issue is with the build pulling in dependencies from the internet at runtime. If we run the docker build as part of a pre-designated step, then it's perfectly fine

Reply:
Yes. I guess for privatecloud as well, the build would happen in the design time, hence, I feel it's ok.